contrib/native/client/src/include/drill/userProperties.hpp

```diff
@@ -28,6 +28,18 @@ class DECLSPEC_DRILL_CLIENT DrillUserProperties{
         static const std::map<std::string, uint32_t> USER_PROPERTIES;
 
         DrillUserProperties(){};
+        
+        /// @brief Update the property value associate with the property key if the value is 
+        /// empty.
+        /// 
+        /// @param in_propName              The property name.
+        /// @param in_propValue             The property value.
+        void setPropertyIfEmpty(const std::string& in_propName, const std::string& in_propValue){
+            // If the element value is empty, then update.
+            if (isPropSet(in_propName) && m_properties[in_propName].empty()){
```

Comment:
Better check would be 
`if ( !isPropSet(in_propName) || m_properties[in_propName].empty() )`

Reply:
certainly